pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/PredicateComparisonRewriter.java

```diff
@@ -33,75 +36,126 @@ public class PredicateComparisonRewriter implements QueryRewriter {
+        switch (filterKind) {
+          case AND:
+          case OR:
+          case NOT:
+            for (int i = 0; i < operands.size(); i++) {
+              Expression operand = operands.get(i);
+              operands.set(i, updatePredicate(operand, function));
             }
             break;
-          }
+          case EQUALS:
+          case NOT_EQUALS:
+          case GREATER_THAN:
+          case GREATER_THAN_OR_EQUAL:
+          case LESS_THAN:
+          case LESS_THAN_OR_EQUAL:
+            Expression firstOperand = operands.get(0);
+            Expression secondOperand = operands.get(1);
 
-          // Handle predicate like 'a > b' -> 'a - b > 0'
-          if (!secondOperand.isSetLiteral()) {
-            Expression minusExpression = RequestUtils.getFunctionExpression("minus");
-            minusExpression.getFunctionCall().setOperands(Arrays.asList(firstOperand, secondOperand));
-            operands.set(0, minusExpression);
-            operands.set(1, RequestUtils.getLiteralExpression(0));
+            // Handle predicate like '10 = a' -> 'a = 10'
+            if (firstOperand.isSetLiteral()) {
+              if (!secondOperand.isSetLiteral()) {
+                function.setOperator(getOppositeOperator(filterKind).name());
+                operands.set(0, secondOperand);
+                operands.set(1, firstOperand);
+              }
+              break;
+            }
+
+            // Handle predicate like 'a > b' -> 'a - b > 0'
+            if (!secondOperand.isSetLiteral()) {
+              Expression minusExpression = RequestUtils.getFunctionExpression("minus");
+              minusExpression.getFunctionCall().setOperands(Arrays.asList(firstOperand, secondOperand));
+              operands.set(0, minusExpression);
+              operands.set(1, RequestUtils.getLiteralExpression(0));
+              break;
+            }
             break;
-          }
-          break;
-        default:
-          int numOperands = operands.size();
-          for (int i = 1; i < numOperands; i++) {
-            if (!operands.get(i).isSetLiteral()) {
-              throw new SqlCompilationException(
-                  String.format("For %s predicate, the operands except for the first one must be literal, got: %s",
-                      filterKind, expression));
+          default:
+            int numOperands = operands.size();
+            for (int i = 1; i < numOperands; i++) {
+              if (!operands.get(i).isSetLiteral()) {
+                throw new SqlCompilationException(String
+                    .format("For %s predicate, the operands except for the first one must be literal, got: %s",
+                        filterKind, expression));
+              }
             }
-          }
-          break;
+            break;
+        }
       }
+    } else if (type == ExpressionType.IDENTIFIER) {
+      expression = convertPredicateToBooleanExpression(expression, parentFunction);
     }
+
```

Comment:
We should also update literal to match the SQL semantic

Reply:
Can we add support for literals as a separate PR - asking because adding support and test for literals will make this PR larger? I'll send out a PR just for literals soon.

Reply:
Sounds good, we may add a TODO here and handle it separately